pydantic/_internal/_generate_schema.py

```diff
@@ -996,14 +996,22 @@ def json_schema_update_func(schema: CoreSchemaOrField, handler: GetJsonSchemaHan
 
         # apply alias generator
         alias_generator = self._config_wrapper.alias_generator
-        if alias_generator and (field_info.alias_priority is None or field_info.alias_priority <= 1):
-            alias = alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
-            field_info.alias = alias
-            field_info.validation_alias = alias
-            field_info.serialization_alias = alias
-            field_info.alias_priority = 1
+        if alias_generator:
+            if field_info.alias_priority is None or field_info.alias_priority <= 1:
+                alias = alias_generator(name)
+                if not isinstance(alias, str):
+                    raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
+                field_info.alias = alias
+                field_info.validation_alias = alias
+                field_info.serialization_alias = alias
+                field_info.alias_priority = 1
+            elif field_info.alias is None:
+                alias = alias_generator(name)
+                if not isinstance(alias, str):
+                    raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
+                field_info.alias = alias
+                field_info.serialization_alias = field_info.serialization_alias or alias
+                field_info.validation_alias = field_info.validation_alias or alias
```

Comment:
I think in principle you can use `field_info.serialization_alias = ''`, and we've had that request before, but with these `or`s that would get replaced. So I think this should be changed to something like:
```python
if field_info.serialization_alias is None:
    field_info.serialization_alias = alias
```
and similar for `validation_alias`.

I recognize you could maybe fit this in one line but I think it gets long enough with an `is None` check instead of just an `or` that it's not worth it any more.